app/Http/Controllers/ExportController.php

```diff
@@ -608,10 +701,12 @@ private function getTierLabelField($guild) {
                     WHEN gi.tier = 5 THEN '{$tiers[5]}'
                     WHEN gi.tier = 6 THEN '{$tiers[6]}'
                 END AS 'item_tier_label'";
-        } else if ($guild->tier_mode == Guild::TIER_MODE_NUM) {
+        }
+
+        if ($guild->tier_mode === Guild::TIER_MODE_NUM) {
             return "gi.tier AS 'item_tier_label'";
-        } else {
-            return "null AS 'item_tier_label'";
         }
+
+        return "null AS 'item_tier_label'";

```

Comment:
I know that logically this functions the same any only ever one will be triggered, but why are we removing the `else if` `else` here? It's personal preference, but I do prefer explicitly keeping return statements inside the `if` statement so the functionality is immediately explicit.

Reply:
For this particular method I completely agree with you, it doesn't matter too much and could be considered personal preference. Then again, if you do it this way here then you'd have to do it everywhere in order to keep a coding standard up, which seems to be the case. And that's also where the problem lies.

Look at the `ItemController@query` method for example where the same thing happens:

![image](https://user-images.githubusercontent.com/4224975/126078812-e9c9b0fc-f06c-4b7b-a4b3-01b101fd3a67.png)

At the end of the `if`-statement you jump back out of the `if`-statement with the transform (top red line) only to end up in the `else` of the parent `if`-statement which is EXTREMELY easy to miss because of the `return` in the anonymous function passed to the transform method. When I glanced over the method I didn't notice that it was in a closure and was confused for a little bit.

Compare that to the refactored "prevent unnecessary indentation" version if you will and hopefully it makes a bit more sense:

![image](https://user-images.githubusercontent.com/4224975/126078897-c9d637ee-4e2d-479d-9acf-86a91b568f15.png)

Also if it helps my case: you use Laravel as your weapon of choice which also lives by the "prevent indentation unless needed" mantra, so mixing the two especially when you start extending core functionalities would got sloppy pretty quickly.

Reply:
Yeah, this is definitely a case where the nested IF statements with returns doesn't work out so well, especially given that the existing code was clearly written but not reviewed for clarity. I was looking online for other people's opinions (not the first time I've looked for this particular use case), and more people do seem to prefer the method you're demonstrating: https://softwareengineering.stackexchange.com/a/18473/90043

Reply:
I'm not super zealous about either one. Probably a case-by-case basis, I agree that the refactor you've demonstrated is easier to follow. The return in the first example (the bad example) should have returned within the `if` statement, but even then it's not nice to have returns nested really deeply, because you now have to read through all the code to understand it.

Reply:
Go for whatever makes sense to you. I'll bring it up in review if I find something hard to digest, but otherwise I trust your judgement.

Reply:
Sounds like a plan!